### ISSUE TITLE
Can only download files from instances post-converge

### ISSUE THREAD

**Issue by jayhendren** [test-kitchen/test-kitchen#1470]
## Description

Per #783 and #1306, support was added for downloading files from instances.  One of the use cases specifically mentioned in #783 was downloading jUnit test reports for parsing by Jenkins.  However, the features added in #1306 do not appear to support this use case.  I can only download files post-converge, not post-verify, which means I cannot download jUnit test reports after `kitchen verify` has run tests.

I tried both of the following in my `.kitchen.yml`; only the latter appears to have any effect:

```
verifier:
  name: cub_busser
  downloads:
    /tmp/verifier/junit: test/junit
```

```
provisioner:
  name: chef_zero
  downloads:
    /tmp/verifier/junit: test/junit
```

## Kitchen Version

Tell us which version of test-kitchen you are using (`kitchen --version`).  

```
[birdsnest ~/Work/git/cookbooks/cub_test](foo|✔)[I]% kitchen --version
Test Kitchen version 1.22.0
```

## ChefDK Version

If you are running test-kitchen via ChefDK, `chef --version` will provide additional relevent version details.

```
[birdsnest ~/Work/git/cookbooks/cub_test](foo|✔)[I]% chef --version
Chef Development Kit Version: 3.1.0
chef-client version: 14.2.0
delivery version: master (6862f27aba89109a9630f0b6c6798efec56b4efe)
berks version: 7.0.4
kitchen version: 1.22.0
inspec version: 2.1.72
```

## Platform Version

Arch Linux

## Replication Case

See above

## Kitchen Output

n/a

## Kitchen Diagnose

n/a

## NOTE:

This repository is for core issues with the Test-Kitchen framework, so if the issue is with a plugin you may be referred to file the issue with the appropriate plugin.

**Comment by coderanger** [comment]
Try something like this as a lifecycle hook:

```
post_verify: kitchen $KITCHEN_INSTANCE exec cat /whatever/junit.xml > junit.xml
```

Might need to play with the quoting but something like that.

**Comment by Poohblah** [comment]
@coderanger I was able find an alternative solution to my particular situation using synced folders that works pretty well.  I tried a lifecycle hook approach very similar to what you suggested as well, but I decided to abandon that because it spit out extra output that corrupted the resulting xml files.  All that is irrelevant to this particular issue, though: I'm filing this issue specifically because a use case for downloads specifically called out in #783 was not addressed with the particular downloads implementation chosen.
